onnxruntime/python/onnxruntime_pybind_exceptions.h

```diff
@@ -0,0 +1,110 @@
+#include <pybind11/pybind11.h>
+#include <stdexcept>
+
+namespace onnxruntime {
+namespace python {
+
+// onnxruntime::python exceptions map 1:1 to onnxruntime:common::StatusCode enum.
+class Fail : public std::runtime_error {
+ public:
+  explicit Fail(const std::string& what) : std::runtime_error(what) {}
+};
+class InvalidArgument : public std::runtime_error {
+ public:
+  explicit InvalidArgument(const std::string& what) : std::runtime_error(what) {}
+};
+class NoSuchFile : public std::runtime_error {
+ public:
+  explicit NoSuchFile(const std::string& what) : std::runtime_error(what) {}
+};
+class NoModel : public std::runtime_error {
+ public:
+  explicit NoModel(const std::string& what) : std::runtime_error(what) {}
+};
+class EngineError : public std::runtime_error {
+ public:
+  explicit EngineError(const std::string& what) : std::runtime_error(what) {}
+};
+class RuntimeException : public std::runtime_error {
+ public:
+  explicit RuntimeException(const std::string& what) : std::runtime_error(what) {}
+};
+class InvalidProtobuf : public std::runtime_error {
+ public:
+  explicit InvalidProtobuf(const std::string& what) : std::runtime_error(what) {}
+};
+class ModelLoaded : public std::runtime_error {
+ public:
+  explicit ModelLoaded(const std::string& what) : std::runtime_error(what) {}
+};
+class NotImplemented : public std::runtime_error {
+ public:
+  explicit NotImplemented(const std::string& what) : std::runtime_error(what) {}
+};
+class InvalidGraph : public std::runtime_error {
+ public:
+  explicit InvalidGraph(const std::string& what) : std::runtime_error(what) {}
+};
+class EPFail : public std::runtime_error {
+ public:
+  explicit EPFail(const std::string& what) : std::runtime_error(what) {}
+};
+
+void RegisterExceptions(pybind11::module& m) {
+  pybind11::register_exception<Fail>(m, "Fail");
+  pybind11::register_exception<InvalidArgument>(m, "InvalidArgument");
+  pybind11::register_exception<NoSuchFile>(m, "NoSuchFile");
+  pybind11::register_exception<NoModel>(m, "NoModel");
+  pybind11::register_exception<EngineError>(m, "EngineError");
+  pybind11::register_exception<RuntimeException>(m, "RuntimeException");
+  pybind11::register_exception<InvalidProtobuf>(m, "InvalidProtobuf");
+  pybind11::register_exception<ModelLoaded>(m, "ModelLoaded");
+  pybind11::register_exception<NotImplemented>(m, "NotImplemented");
+  pybind11::register_exception<InvalidGraph>(m, "InvalidGraph");
+  pybind11::register_exception<EPFail>(m, "EPFail");
+}
+#define ORT_PYBIND_THROW_IF_ERROR(expr)     \
```

Comment:
Could this be turned into an inline function instead of a macro?

Reply:
done.